src/connector/src/source/kinesis/source/message.rs

```diff
@@ -14,16 +14,25 @@
 
 use aws_sdk_kinesis::types::Record;
 use aws_smithy_types_convert::date_time::DateTimeExt;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
 
 use crate::source::{SourceMessage, SourceMeta, SplitId};
 
 #[derive(Clone, Debug)]
 pub struct KinesisMeta {
-    // from `approximate_arrival_timestamp` of type `Option<aws_smithy_types::DateTime>`
-    #[expect(dead_code)]
     timestamp: Option<i64>,
 }
 
+impl KinesisMeta {
+    pub fn extract_timestamp(&self) -> Option<DatumRef<'_>> {
+        self.timestamp.map(|ts| {
+            Some(ScalarRefImpl::Timestamptz(
+                risingwave_common::cast::i64_to_timestamptz(ts).unwrap(),
```

Comment:
Use `Timestamptz::from_micros` when the unit is known. The `i64_to_timestamptz` function will guess the unit.